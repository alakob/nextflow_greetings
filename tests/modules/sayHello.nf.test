nextflow_process {

    name "Test Process sayHello"
    script "modules/sayHello.nf"
    process "sayHello"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = "Greetings"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
