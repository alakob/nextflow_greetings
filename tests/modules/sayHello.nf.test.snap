{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "Greetings.output.txt:md5,66838632dd4046205cff7e68b5112d46"
                ],
                "greeting_file": [
                    "Greetings.output.txt:md5,66838632dd4046205cff7e68b5112d46"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-26T13:47:34.68808"
    }
}